### PR TITLE
Fixed when invalid workspace was provided

### PR DESCRIPTION
**Pull request by Raalsky**

### CODE REVIEW

File: neptune/new/internal/backends/hosted_neptune_backend.py [hunk 1]
```diff
@@ -207,12 +207,20 @@ def get_project(self, project_id: str) -> Project:
                 raise NeptuneLegacyProjectException(project_id)
             return Project(project.id, project.name, project.organizationName)
         except HTTPNotFound:
+            available_workspaces = self.get_available_workspaces()
+
+            if workspace and not any(
+                filter(lambda aw: aw.name == workspace, available_workspaces)
```
Comment (aniezurawski) [inline neptune/new/internal/backends/hosted_neptune_backend.py:213]: `any()` takes an iterable of booleans, doesn't it? So i think it should be either:
```any(map(lambda aw: aw.name == workspace, available_workspaces)```
or
```list(filter(lambda aw: aw.name == workspace, available_workspaces)```

Your code is correct as long as `workspace` evaluates to True (is not empty and is not None). But it's weird to read.

File: neptune/new/internal/backends/hosted_neptune_backend.py [hunk 2]
```diff
@@ -207,12 +207,20 @@ def get_project(self, project_id: str) -> Project:
...
+            ):
+                # Could not found specified workspace, forces listing all projects
+                workspace = None
+            else:
+                available_workspaces = list()
```
Comment (aniezurawski) [inline neptune/new/internal/backends/hosted_neptune_backend.py:218]: Why reset this list if workspace is None?
Can we just skip fetching available worksapces?

**Review by aniezurawski — CHANGES_REQUESTED** [review]

File: neptune/new/internal/backends/hosted_neptune_backend.py (hunk 1 above)
Comment (Raalsky) [reply neptune/new/internal/backends/hosted_neptune_backend.py:213]: Ok, list was already used in a few lines above :wink:

Comment (pkasprzyk) [reply neptune/new/internal/backends/hosted_neptune_backend.py:213]: and how would you like `next(filter(lambda aw: aw.name == workspace, available_workspaces), None) is None`? :D

Comment (Raalsky) [reply neptune/new/internal/backends/hosted_neptune_backend.py:213]: Nah, looks like a Python 2.x code :P

**Review by pkasprzyk — APPROVED** [review]

**Review by aniezurawski — APPROVED** [review]

**Pull request merged**